ansible/library/osp_copy_vm_disk_to_image.py

```diff
@@ -0,0 +1,195 @@
+# Copyright: (c) 2021, Alberto Gonzalez <alberto.gonzalez@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+import datetime
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: osp_copy_vm_disk_to_image
+short_description: converts a ephemeral VM disk to a image directly with Ceph
+version_added: "2.7"
+description:
+    - this module connects to Ceph and creates an OSP image cloning the ceph image
+options:
+    - items: a list of dictionary (src, dest)
+extends_documentation_fragment:
+    - openstack
+equirements:
+    - "python >= 2.7"
+    - "openstacksdk"
+    - "ibm-cos-sdk"
+author:
+    - Alberto Gonzalez
+'''
+
+EXAMPLES = '''
+   - name: Upload vms from project
+     become: true
+     environment:
+       OS_AUTH_URL: "{{ osp_auth_url }}"
+       OS_USERNAME: "{{ osp_auth_username }}"
+       OS_PASSWORD: "{{ osp_auth_password }}"
+       OS_PROJECT_NAME: "{{ osp_project | default('admin') }}"
+       OS_PROJECT_DOMAIN_ID: "{{ osp_auth_project_domain }}"
+       OS_USER_DOMAIN_NAME: "{{ osp_auth_user_domain }}"
+       OS_INTERFACE: "{{ osp_interface | default('internal') }}"
+       PATH: "/root/.local/bin:{{ ansible_env.PATH }}"
+       CEPH_CONF: "/etc/ceph/{{ ceph_cluster | default('red') }}.conf"
+       osp_copy_vm_disk_to_image: 
+         project: "{{ osp_project }}"
+         ceph_cluster: "{{ ceph_cluster | default('red') }}"
+         items:
+           - src: myvm1
+             dest: myvm1-image
+           - src: myvm2
+             dest: myvm2-image
+'''
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.openstack import openstack_full_argument_spec, openstack_module_kwargs, \
+    openstack_cloud_from_module
```

Comment:
I wonder if it is appropriate to use the openstacak module utils in Ansible core now that this has moved to a collection?
Should we develop against the collection style implementation, for example:
https://github.com/openstack/ansible-collections-openstack/blob/master/plugins/modules/image.py#L158

Reply:
changed, thanks